### PR TITLE
Run CKAN container when db is ready

### LINKED ISSUE #3075: `docker-compose up` does not work
The way that you are specifying environmental variables in these lines is incorrect: https://github.com/ckan/ckan/blob/master/contrib/docker/my_init.d/50_configure#L36-L49

As a result the CKAN container cannot connect to the DB or SOLR

``` bash
ckan | *** Running /etc/my_init.d/50_configure...
ckan | /etc/my_init.d/50_configure: 37: /etc/my_init.d/50_configure: DB_ENV_POSTGRES_USER: parameter not set
ckan | no DATABASE_URL specified and linked container called 'db' was not found
ckan | *** /etc/my_init.d/50_configure failed with status 1
ckan |
ckan | *** Killing all processes...
ckan exited with code 1
```

I added the ENV's for `DATABASE_URL` and `SOLR_URL` directly into docker-compose.yml and everything works.

``` yml
version: '2'
services:
  ckan:
    container_name: ckan
    image: ckan/ckan:latest
    depends_on:
      - db
      - solr
    links:
       - db
       - solr
    ports:
       - "80:80"
    environment:
       DATABASE_URL: "postgresql://ckan:ckan@db:5432/ckan"
       SOLR_URL: "http://solr:8983/solr/ckan"
  db:
    container_name: db
    image: ckan/postgresql:latest
  solr:
    container_name: solr
    image: ckan/solr:latest
```

My Docker version info:

``` bash
$ docker version                                                                                                                      [16:37:48]
Client:
 Version:      1.10.2
 API version:  1.22
 Go version:   go1.5.3
 Git commit:   c3959b1
 Built:        Mon Feb 22 22:37:33 2016
 OS/Arch:      darwin/amd64

Server:
 Version:      1.11.1
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   5604cbe
 Built:        Wed Apr 27 00:34:20 2016
 OS/Arch:      linux/amd64
```

### LINKED ISSUE #3319: docker-compose up will fail on initial database timeout
### CKAN Version if known (or site URL)
2.6.0 (docker images tagged dev-v2.6)

### Please describe the expected behaviour
With contrib/docker/docker-compose.yml modified to add required environment variables (#3075)
'docker-compose up' should bring up a working system

### Please describe the actual behaviour
The 3 containers all appear to be able to start. The use of depends_on starts them in the correct order.

However, the db container has not fully started by the time the ckan container looks for it.

It fails with 
```
ckan    | sqlalchemy.exc.OperationalError: (OperationalError) could not connect to server: Connection refused
ckan    | 	Is the server running on host "db" (172.18.0.3) and accepting
```

### What steps can be taken to reproduce the issue? 
```
$ cd contrib/docker
$ docker-compose up
```

### PR DESCRIPTION
**Pull request by frafra**
Fixes #3075
Fixes #3319

### Proposed fixes:

Docker allows to do health checks inside docker-compose.yml files since 2.1. This allows to specify when to run a container based on the health status of its dependencies. With this patch CKAN container is started only when the PostgreSQL database is ready.
Current docker-compose.yml version is 3.0, so there is no compatibility issue (docker-compose.yml 3.0 requires Docker Engine 1.13.0+, while 2.1 needs 1.12.0+).

### Features:

- [ ] includes tests covering changes
- [ ] includes updated documentation
- [ ] includes user-visible changes
- [ ] includes API changes
- [ ] includes bugfix for possible backport

### CODE REVIEW

**Comment by frafra** [comment]
I just fixed my email address because I did it at work. Nothing else changed.

File: contrib/docker/docker-compose.yml [hunk 1]
```diff
@@ -58,6 +61,8 @@ services:
       - POSTGRES_PASSWORD=${POSTGRES_PASSWORD}
     volumes:
       - pg_data:/var/lib/postgresql/data
+    healthcheck:
+      test: ["CMD", "pg_isready", "-U", "postgres"]
```
Comment (wardi) [inline contrib/docker/docker-compose.yml]: TIL `pg_isready` is a thing. Cool!

Comment (frafra) [reply contrib/docker/docker-compose.yml]: It is :-)

**Comment by frafra** [comment]
I tested a lot of times during the last couple of weeks, and I got consistent results. This is useful for monitoring in my opinion and in the future we could extend it to other containers. For the moment, I suppose that fixing the "you-have-to-start-ckan-twice" bug is the priority, so I kept the pull request small.

Let me know if there is anything else I can do about it.

**Comment by frafra** [comment]
This method is not compatible with docker compose 3.0 or greater.

**Pull request merged**